pkg/licensing/expression/category.go

```diff
@@ -358,4 +365,131 @@ var (
 		Unlicense,
 		ZeroBSD,
 	}
+
+	// SpdxLicenseExceptions contains all supported SPDX Exceptions
+	// cf. https://github.com/spdx/license-list-data/blob/592c2dcb8497c6fe829eea604045f77d3bce770b/json/exceptions.json
+	// used `awk -F'"' '/"licenseExceptionId":/ {print toupper("\"" $4 "\"," )}' exceptions.json ` command
+	spdxLicenseExceptions = []string{
+		"389-EXCEPTION",
+		"ASTERISK-EXCEPTION",
+		"ASTERISK-LINKING-PROTOCOLS-EXCEPTION",
+		"AUTOCONF-EXCEPTION-2.0",
+		"AUTOCONF-EXCEPTION-3.0",
+		"AUTOCONF-EXCEPTION-GENERIC",
+		"AUTOCONF-EXCEPTION-GENERIC-3.0",
+		"AUTOCONF-EXCEPTION-MACRO",
+		"BISON-EXCEPTION-1.24",
+		"BISON-EXCEPTION-2.2",
+		"BOOTLOADER-EXCEPTION",
+		"CGAL-LINKING-EXCEPTION",
+		"CLASSPATH-EXCEPTION-2.0",
+		"CLISP-EXCEPTION-2.0",
+		"CRYPTSETUP-OPENSSL-EXCEPTION",
+		"DIGIRULE-FOSS-EXCEPTION",
+		"ECOS-EXCEPTION-2.0",
+		"ERLANG-OTP-LINKING-EXCEPTION",
+		"FAWKES-RUNTIME-EXCEPTION",
+		"FLTK-EXCEPTION",
+		"FMT-EXCEPTION",
+		"FONT-EXCEPTION-2.0",
+		"FREERTOS-EXCEPTION-2.0",
+		"GCC-EXCEPTION-2.0",
+		"GCC-EXCEPTION-2.0-NOTE",
+		"GCC-EXCEPTION-3.1",
+		"GMSH-EXCEPTION",
+		"GNAT-EXCEPTION",
+		"GNOME-EXAMPLES-EXCEPTION",
+		"GNU-COMPILER-EXCEPTION",
+		"GNU-JAVAMAIL-EXCEPTION",
+		"GPL-3.0-389-DS-BASE-EXCEPTION",
+		"GPL-3.0-INTERFACE-EXCEPTION",
+		"GPL-3.0-LINKING-EXCEPTION",
+		"GPL-3.0-LINKING-SOURCE-EXCEPTION",
+		"GPL-CC-1.0",
+		"GSTREAMER-EXCEPTION-2005",
+		"GSTREAMER-EXCEPTION-2008",
+		"HARBOUR-EXCEPTION", //nolint:misspell
+		"I2P-GPL-JAVA-EXCEPTION",
+		"INDEPENDENT-MODULES-EXCEPTION",
+		"KICAD-LIBRARIES-EXCEPTION",
+		"LGPL-3.0-LINKING-EXCEPTION",
+		"LIBPRI-OPENH323-EXCEPTION",
+		"LIBTOOL-EXCEPTION",
+		"LINUX-SYSCALL-NOTE",
+		"LLGPL",
+		"LLVM-EXCEPTION",
+		"LZMA-EXCEPTION",
+		"MIF-EXCEPTION",
+		"MXML-EXCEPTION",
+		"NOKIA-QT-EXCEPTION-1.1",
+		"OCAML-LGPL-LINKING-EXCEPTION",
+		"OCCT-EXCEPTION-1.0",
+		"OPENJDK-ASSEMBLY-EXCEPTION-1.0",
+		"OPENVPN-OPENSSL-EXCEPTION",
+		"PCRE2-EXCEPTION",
+		"PS-OR-PDF-FONT-EXCEPTION-20170817",
+		"QPL-1.0-INRIA-2004-EXCEPTION",
+		"QT-GPL-EXCEPTION-1.0",
+		"QT-LGPL-EXCEPTION-1.1",
+		"QWT-EXCEPTION-1.0",
+		"ROMIC-EXCEPTION",
+		"RRDTOOL-FLOSS-EXCEPTION-2.0",
+		"SANE-EXCEPTION",
+		"SHL-2.0",
+		"SHL-2.1",
+		"STUNNEL-EXCEPTION",
+		"SWI-EXCEPTION",
+		"SWIFT-EXCEPTION",
+		"TEXINFO-EXCEPTION",
+		"U-BOOT-EXCEPTION-2.0",
+		"UBDL-EXCEPTION",
+		"UNIVERSAL-FOSS-EXCEPTION-1.0",
+		"VSFTPD-OPENSSL-EXCEPTION",
+		"WXWINDOWS-EXCEPTION-3.1",
+		"X11VNC-OPENSSL-EXCEPTION",
+	}
```

Comment:
There are cases when users run multiple times.
Downloading these files for each run is not good.
But we can save `licenses.json` and `exceptions.json` files in the cache dir and use them.
The files contain `releaseDate` field, so we can update this file only when `releaseDate + 3 months` has expired.

> The license list is updated every 3 months

How strictly is this rule followed?

Anyway let's move this discussion into another issue/pr.

___ 

> What I do in the code I maintain is attempt to access the current JSON files on the website https://spdx.org/licenses/licenses.json and https://spdx.org/licenses/exceptions.json

I found that https://spdx.org/licenses/exceptions.json and https://github.com/spdx/license-list-data/blob/592c2dcb8497c6fe829eea604045f77d3bce770b/json/exceptions.json are different (see `harbour-exception`).
Which file would be more correct to use?

Reply:
> How strictly is this rule followed?

Not very strictly.  There is, however, a license list version field which is reliably incremented on release.

> I found that https://spdx.org/licenses/exceptions.json and https://github.com/spdx/license-list-data/blob/592c2dcb8497c6fe829eea604045f77d3bce770b/json/exceptions.json are different (see harbour-exception).
Which file would be more correct to use?

The lists at https://spdx.org/licenses - these will always be the latest _released_ version.  The github repo master will have the latest in development version which may not be stable.  The github repo is tagged with release versions, so if you go to the tag for the latest release in github, it will match what is on the website.

Reply:
Okay, update exception list in https://github.com/aquasecurity/trivy/pull/8077/commits/659f992f9eccbfec55ea65fe7b2bdc973d892c8e